packages/block-editor/src/components/inserter/menu.js

```diff
@@ -49,9 +50,15 @@ function InserterMenu(
 	},
 	ref
 ) {
+	// Used for the search field only.
 	const [ filterValue, setFilterValue ] = useState(
 		__experimentalFilterValue
 	);
+	// Used for the rest of the inserter, debounced for performance reasons.
+	const [ delayedFilterValue, setDelayedFilterValue ] = useState(
+		__experimentalFilterValue
+	);
+	const debouncedSetFilterValue = useDebounce( setDelayedFilterValue, 100 );
```

Comment:
@ntsekouras I think you did something similar in a recent PR right? just to make sure we do the same in both.

Reply:
Actually I have duplicated a util hook `useDebouncedInput` that lives in two packages: https://github.com/WordPress/gutenberg/search?q=useDebouncedInput. Maybe it's useful to create a util for all packages if it makes sense, but for now I think  we could just move the `block-editor` one from [media-tab/hooks](https://github.com/WordPress/gutenberg/blob/9a4146f3a64184f904d9631c92d2f134957de318/packages/block-editor/src/components/inserter/media-tab/hooks.js#L14) and reuse it here. What do you think?

Reply:
Sounds good, can't think of a good reason to not reuse similar prior art 👍